content/posts/why-we-use-cucumber-for-end-to-end-testing/index.md

```diff
@@ -0,0 +1,78 @@
++++
+date = "2021-12-03T14:37:23+07:00"
+author = "vctqs1"
+description = "What is the purpose of Cucumber and End-to-end testing? Why do we need to invest in them?"
+title = "Why we use Cucumber for end-to-end testing?"
+categories = ["Automation", "End-to-end", "BDD"]
+tags = ["bdd", "end-to-end", "cucumber", "automation", "test"]
+slug = "why-we-use-cucumber-for-end-to-end-testing"
++++
+
+# Why we use Cucumber for end-to-end testing?
+
+Before knowing why we use cucumber for End-to-end testing. Let understand what is end-to-end testing and cucumber
+
+## What is End to end testing?
+
+End to end testing (E2E testing) is a software testing method that validates entire software from beginning to end.
+
+The purpose of end-to-end testing is to simulate what a real user scenario and testing whole software for dependencies, data integrity and communication with other systems, interfaces, network connectivity and databases to exercise complete production like scenario.
+
+## What is Cucumber?
+
+What is the world said:
+
+> Cucumber is a software tool that supports behavior-driven development (BDD).Central to the Cucumber BDD approach is its ordinary language parser called Gherkin. It allows expected software behaviors to be specified in a logical language that customers can understand. As such, Cucumber allows the execution of feature documentation written in business-facing text. It is often used for testing other software. It runs automated acceptance tests written in a behavior-driven development (BDD) style.
+
+### What is Behavior-driven development?
+
+In software engineering, behavior-driven development (BDD) is an agile software development process that encourages collaboration among developers, quality assurance testers, and customer representatives in a software project.
```

Comment:
In software engineering, behavior-driven development (BDD) is an agile software development process which encourage collaboration among developers, QAs (Quality Assurance team), and customer representatives in a project.